### PR TITLE
Voxel traversal

### PR DESCRIPTION
**Pull request by roxxel**

### CODE REVIEW

File: Obsidian.API/_Types/Traversal.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+            float tMaxX, tMaxY, tMaxZ, tDeltaX, tDeltaY, tDeltaZ;
+            VectorF voxel = new VectorF();
+
+            float x1 = StartPoint.X, y1 = StartPoint.Y, z1 = StartPoint.Z; // start point   
+            float x2 = EndPoint.X, y2 = EndPoint.Y, z2 = EndPoint.Z; // end point   
+
+
+            var dx = Math.Sign(x2 - x1); // x direction
+            if (dx != 0) tDeltaX = Math.Min(dx / (x2 - x1), 10000000.0f); else tDeltaX = 10000000.0f;
```
Comment (Seb-stian) [inline Obsidian.API/_Types/Traversal.cs]: `10000000.0f` seems to be a magic number. Could it be `float.MaxValue` or `float.PositiveInfinity`?

Comment (roxxel) [reply Obsidian.API/_Types/Traversal.cs]: Done

File: Obsidian/Entities/Entity.cs [hunk 2]
```diff
@@ -28,22 +28,6 @@ public class Entity : IEquatable<Entity>, IEntity
         public Angle Pitch { get; set; }
 
         public Angle Yaw { get; set; }
-
```
Comment (Naamloos) [inline Obsidian/Entities/Entity.cs:31]: Why exactly is this removed?

File: Obsidian.API/_Interfaces/IEntity.cs [hunk 3]
```diff
@@ -8,7 +8,6 @@ public interface IEntity
         public VectorF Position { get; set; }
         public Angle Pitch { get; set; }
         public Angle Yaw { get; set; }
-        public VectorF LookDirection { get; }
```
Comment (Naamloos) [inline Obsidian.API/_Interfaces/IEntity.cs:11]: same here

Comment (roxxel) [reply Obsidian.API/_Interfaces/IEntity.cs:11]: Didn't work as expected

File: Obsidian/Entities/Entity.cs (hunk 2 above)
Comment (roxxel) [reply Obsidian/Entities/Entity.cs:31]: Just misclick

**Comment by TheBunnies** [comment]
Sooo, I don't know if my opinion is matter here. But I have a few suggestions about this PR. First of all I don't understand why you're combining **var's** with explicit types, sometimes you write **List<VectorF> blocks = new()** and sometimes you write **var something = new()**. Second, why you use IEnumerable as a return object? IEnumerable makes a lot of sense in the method arguments but as a return type it adds more implicity in my opnion.

**Comment by Naamloos** [comment]
@TheBunnies Thank you for your review! Anyone's opinion matters if what they say is relevant and truthful!

**Pull request merged**